multimodal/src/autogluon/multimodal/utils/object_detection.py

```diff
@@ -658,12 +729,119 @@ def get_coco_format_classes(sample_data_path):
+    obj_ids = []
+    conf_scores = []
+    detection_classes = get_detection_classes(test_path)
+    idx2classname = {i: classname for (i, classname) in enumerate(detection_classes)}
+    for idx, per_cls_bboxes in enumerate(image_pred):
+        for bbox in per_cls_bboxes:
+            ## x1, y1, x2, y2, conf_score
+            if bbox[4] > 0.3:
+                tlwhs.append(bbox_xyxy_to_xywh(list(bbox[:4])))
+                obj_ids.append(idx)
+                conf_scores.append(bbox[4])
+
+    visualized_im = plot_tracking(im, tlwhs, obj_ids, idx2classname, scores=conf_scores)
+    imgname = os.path.basename(image_path)
+    if not os.path.exists(visualization_result_dir):
+        os.makedirs(visualization_result_dir, exist_ok=True)
+    cv2.imwrite(os.path.join(visualization_result_dir, imgname), visualized_im)
+    return visualized_im
+
+
+def plot_tracking(image, tlwhs, obj_ids, idx2classname, scores=None, frame_id=0, fps=0., ids2=None, line_thickness_unmatched=None,
+                  line_thickness=None):
+    try:
+        import cv2
+    except:
+        raise ImportError("No module named: cv2. Please install cv2 by 'pip install cv2'")
+    im = np.ascontiguousarray(np.copy(image))
+    im_h, im_w = im.shape[:2]
+
+    top_view = np.zeros([im_w, im_w, 3], dtype=np.uint8) + 255
+
+    text_scale = max(1, image.shape[1] / 1600.)
+    text_thickness = 1 if text_scale > 1.1 else 1
+    # line_thickness = max(1, int(image.shape[1] / 500.)) if line_thickness_unmatched is None else line_thickness_unmatched
+    line_thickness = line_thickness if line_thickness_unmatched is None else line_thickness_unmatched
+
+    radius = max(5, int(im_w / 140.))
+    if fps != -1:
+        cv2.putText(im, 'frame: %d fps: %.2f num: %d' % (frame_id, fps, len(tlwhs)),
+                    (0, int(15 * text_scale)), cv2.FONT_HERSHEY_PLAIN, text_scale, (0, 0, 255), thickness=2)
+    else:
+        cv2.putText(im, 'frame: %d num: %d' % (frame_id, len(tlwhs)),
+                    (0, int(15 * text_scale)), cv2.FONT_HERSHEY_PLAIN, text_scale, (0, 0, 255), thickness=2)
+
+    for i, tlwh in enumerate(tlwhs):
+        x1, y1, w, h = tlwh
+        intbox = tuple(map(int, (x1, y1, x1 + w, y1 + h)))
+        obj_id = int(obj_ids[i])
+        id_text = idx2classname[obj_ids[i]]
+        if ids2 is not None:
+            id_text = id_text + ', {}'.format(int(ids2[i]))
+        if scores is not None:
+            id_text = id_text + ', {:.3f}'.format(float(scores[i]))
+        _line_thickness = 1 if obj_id <= 0 else line_thickness
+        color = get_color(abs(obj_id))
+        cv2.rectangle(im, intbox[0:2], intbox[2:4], color=color, thickness=line_thickness)
+        cv2.putText(im, id_text, (intbox[0], intbox[1] + 30), cv2.FONT_HERSHEY_PLAIN, text_scale, (0, 0, 255),
+                    thickness=text_thickness)
+    return im
+
+
+def get_color(idx):
+    idx = idx * 3
+    color = ((37 * idx) % 255, (17 * idx) % 255, (29 * idx) % 255)
+    return color
```

Comment:
newline at the end of the file

Reply:
Fixed in 8a425d699a934c23056f697a2145eec7162e15d5